### PR TITLE
NEW: Add `findIndex` function and tests

### PR DESCRIPTION
**Pull request by r-hyndman**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,11 @@
-// export your functions here
+/* eslint-disable require-jsdoc */
+function findIndex(array, test) {
```
Comment (reubsinit) [inline src/index.js]:
```suggestion
function findIndex(array, callback) {
```

File: src/index.js [hunk 2]
```diff
@@ -1 +1,11 @@
...
+  for (let i = 0; array.length; ++i) {
+    if (test(array[i], i, array)) return i;
```
Comment (reubsinit) [inline src/index.js]:
```suggestion
    if (callback(array[i], i, array)) return i;
```

File: tests/findIndex.spec.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {findIndex} from '../src';
+
+/* eslint-disable require-jsdoc */
+const testArray = [5, 12, 8, 130, 44];
+
+describe('findIndex', () => {
```
Comment (reubsinit) [inline tests/findIndex.spec.js:6]: I reckon it's probably worth defining a spec that doesn't find an element in the test data

**Review by reubsinit — CHANGES_REQUESTED** [review]
Looking good! 3 changes requested and a rebase against `main`.

File: tests/findIndex.spec.js (hunk 3 above)
Comment (reubsinit) [reply tests/findIndex.spec.js:6]: Thoughts on this?

Comment (r-hyndman) [reply tests/findIndex.spec.js:6]: Shit, sorry. I did change it, but it must have been doofed on the rebase

Comment (r-hyndman) [reply tests/findIndex.spec.js:6]: Oh never mind, it was changed :) I changed the test to > 100 and the data value 130 to 90

**Review by reubsinit — APPROVED** [review]

**Pull request merged**